src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -914,5 +983,32 @@ private static string CreateDeterminismKey(CommandLineArguments args, string[] r
 
             return builder.ToString();
         }
+
+
+        /// <summary>
+        /// The set of source file paths that are in the set of embedded paths.
+        /// This is used to prevent reading source files that are embedded twice.
+        ///
+        /// It further prevents a race where the file we embed could have changed
+        /// since the time it was parsed.
+        /// </summary>
+        protected IReadOnlySet<string> EmbeddedSourcePaths { get; }
```

Comment:
>protected IReadOnlySet<string> EmbeddedSourcePaths { get; } [](start = 8, length = 59)

Move to the top of the class decl, we usually group state members (field/autoproperties) at the top.